jaxrs-api/src/test/java/javax/ws/rs/core/EntityTagTest.java

```diff
@@ -0,0 +1,70 @@
+package javax.ws.rs.core;
+
+import static org.hamcrest.CoreMatchers.*;
+import static org.junit.Assert.*;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.mockito.Mockito.*;
+
+import javax.ws.rs.ext.RuntimeDelegate;
+
+public class EntityTagTest {
+
+    private EntityTag tag1;
+    private EntityTag tag2;
+    private EntityTag tag3;
+    private EntityTag tag4;
+
+
+    @Before
+    public void setUp() {
+        RuntimeDelegate.setInstance(mock(RuntimeDelegate.class));
+
+        tag1 = new EntityTag("value1", true);
+        tag2 = new EntityTag("value1", true);
+        tag3 = new EntityTag("value2", true);
+        tag4 = new EntityTag("value1", false);
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        RuntimeDelegate.setInstance(null);
+    }
+
+    @Test
+    public void shouldBeEqualToTheSameInstance() {
+        assertThat(tag1, equalTo(tag1));
+        assertThat(tag1.hashCode(), equalTo(tag1.hashCode()));
+    }
+
+    @Test
+    public void shouldBeEqualsForSameFieldValues() {
+        assertThat(tag1, equalTo(tag2));
+        assertThat(tag1.hashCode(), equalTo(tag2.hashCode()));
+    }
+
+    @Test
+    public void shouldNotBeEqualIfValueFieldDiffers() {
+        assertThat(tag1, not(equalTo(tag3)));
+        assertThat(tag1.hashCode(), not(equalTo(tag3.hashCode())));
+    }
+
+    @Test
+    public void shouldNotBeEqualIfWeekSettingDiffers() {
```

Comment:
I wonder if we really want to have negative tests? I mean, there could happen a lot of things we do not want to happen, but do we have to test them?

Reply:
IMO this test makes sense. So let's keep it. Or what kind of downside do you see if we keep it?

Reply:
See comment below: They are actually there to get to 100% branch coverage.